executor/join.go

```diff
@@ -293,23 +289,32 @@ func (e *HashJoinExec) joinOneBigRow(ctx *hashJoinCtx, bigRow *Row) bool {
 	if e.bigFilter != nil {
 		bigMatched, err = expression.EvalBool(ctx.bigFilter, bigRow.Data, e.ctx)
 		if err != nil {
-			e.resultErr <- errors.Trace(err)
+			e.resultCh <- &execResult{err: errors.Trace(err)}
 			return false
 		}
 	}
 	if bigMatched {
 		matchedRows, err = e.constructMatchedRows(ctx, bigRow)
 		if err != nil {
-			e.resultErr <- errors.Trace(err)
+			e.resultCh <- &execResult{err: errors.Trace(err)}
 			return false
 		}
 	}
+	maxRowsCnt := 1000
+	result := &execResult{rows: make([]*Row, 0, maxRowsCnt)}
```

Comment:
This optimization does not make much difference because there is only one bigRow.
We can create `execResult` in `runJoinWorker`, pass it in this function, then append joined rows.